src/client.ts

```diff
@@ -131,6 +125,11 @@ export function createServiceProxy<T>(
 
         return (input: unknown) =>
           Effect.gen(function* () {
+            // Load metadata dynamically
+            const metadata = yield* Effect.promise(() =>
+              loadServiceMetadata(normalizedServiceName),
+            );
```

Comment:
Accept this as a parameter and update code generation of `${service}/index.ts` to pass it in and then tree shaking will work